scss/utilities/_utilities.scss

```diff
@@ -0,0 +1,79 @@
+// Property utilities generation
+
+
+// Loop over every breakpoint
+@each $breakpoint in map-keys($grid-breakpoints) {
+  // Generate media query if needed
+  @include media-breakpoint-up($breakpoint) {
+    $infix: breakpoint-infix($breakpoint, $grid-breakpoints);
+    // Loop over each utility property
+    @each $key, $utility in $utilities {
+
+      // Only render if responsive media queries are enabled or if it's the base media query
+      @if type_of($utility) == "map" and (map_get($utility, responsive) or $infix == "") {
+        $values: map_get($utility, values);
+
+        // If the values are a list or string, convert it into a map
+        @if type_of($values) == "string" {
+          $values: zip($values, $values);
+        } @else if type_of(nth($values, 1)) != "list" {
+          $values: zip($values, $values);
+        }
+
+        @each $value in $values {
+          $properties: map_get($utility, property);
+
+
+          @if type_of($properties) == "string" {
```

Comment:
What's this string check for?

Reply:
Some utility groups have a single property: (string)
https://github.com/twbs/bootstrap/blob/b1a7b1a335e813bc030b4fa4918f9b826445f9e2/scss/_utilities.scss#L323
others have multiple properties: (list)
https://github.com/twbs/bootstrap/blob/b1a7b1a335e813bc030b4fa4918f9b826445f9e2/scss/_utilities.scss#L433